### ISSUE TITLE
Python class only with async methods support

### ISSUE THREAD

**Issue by yosshy** [doxygen/doxygen#8336]
**Describe the bug**
Doxygen doesn't generate a class reference of python properly if the class had multiple async methods and no sync method. It generates one with ONLY async method name.

**Expected behavior**
Doxygen generates a class reference with multiple async method names properly.

**To Reproduce**

1. Create a working directory and enter it with cd command.
2. Put a sample source code like below:
    ```
    ## Documentation for an async class.
    #  More details.
    class AsyncPyClass:
    
        ## Documentation for a method_a.
        #  @param self The object pointer.
        async def method_a(self):
            pass
    
        ## Documentation for a method_b.
        #  @param self The object pointer.
        async def method_b(self):
            pass
    
        ## Documentation for a method_c.
        #  @param self The object pointer.
        async def method_c(self):
            pass
    ```
3. Run "doxygen -g Doxyfile" to create a configuration file.
4. Run "doxygen Doxyfile" to generate documents.

**Version**
1.9.2 (31709395a2ab12ba137e87cf4d8b70f2cada8396*)

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
